src/overlays/actors/ovl_Obj_Driftice/z_obj_driftice.c

```diff
@@ -28,52 +37,430 @@ const ActorInit Obj_Driftice_InitVars = {
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80A676F4[] = {
+static f32 D_80A67620[][3] = {
+    { 1.0f, 150.0f, 150.0f },
+    { 1.5f, 250.0f, 200.0f },
+    { 2.0f, 350.0f, 250.0f },
+};
+
+static f32 D_80A67644[] = { 1.5f, 3.0f, 4.5f, 6.0f, 7.5f, 9.0f, 10.5f, 0.0f };
+
+static f32 D_80A67664[][4] = {
+    { 0.6f, 0.5f, 0.2f, 0.4f },
+    { 0.8f, 0.7f, 0.35f, 0.7f },
+    { 1.0f, 1.0f, 0.8f, 1.0f },
+};
+
+typedef struct {
+    /* 0x0 */ s16 unk_00;
+    /* 0x2 */ s16 unk_02;
+    /* 0x4 */ f32 unk_04;
+    /* 0x8 */ f32 unk_08;
+} ObjDrifticeDataStruct; // size = 0xC
+
+static ObjDrifticeDataStruct D_80A67694[] = {
+    { 0x12C, 0x64, 1.3f, 2.0f },
+    { 0x258, 0x1F4, 1.2f, 1.8f },
+    { 0x4B0, 0x3E8, 1.1f, 1.5f },
+};
+
+static ObjDrifticeDataStruct D_80A676B8[] = { { 0x320, 0x190, 200.0f, 400.0f }, { 0x5DC, 0x258, 100.0f, 300.0f } };
+
+static ObjDrifticeDataStruct D_80A676D0[] = {
+    { 0x12C, 0x64, 300.0f, 700.0f },
+    { 0x258, 0x258, 200.0f, 400.0f },
+    { 0x4B0, 0x4B0, 100.0f, 300.0f },
+};
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_STOP),
 };
 
-#endif
+void func_80A66570(ObjDriftice* this, s32 arg1) {
+    Math_Vec3s_ToVec3f(&this->dyna.actor.world.pos, &this->unk_16C[arg1]);
+}
+
+void func_80A665AC(s16* arg0, s16 arg1) {
+    if (arg1 < *arg0) {
+        *arg0 = arg1;
+    } else if (*arg0 < -arg1) {
+        *arg0 = -arg1;
+    }
+}
+
+f32 func_80A665EC(ObjDrifticeStruct* s, ObjDriftice* this) {
+    f32 phi_f2 = 0.0f;
+    f32 sp20;
+
+    if (this->unk_248 >= 10) {
+        sp20 = -8.0f;
+    } else if (this->unk_248 < -9) {
+        sp20 = 0.0f;
+    } else if (this->unk_248 > 0) {
+        sp20 = (Math_CosS(fabsf(this->unk_248) * (0x8000 * 0.1f)) - 1.0f) * 4.0f;
+    } else {
+        sp20 = (Math_CosS(fabsf(this->unk_248) * (0x8000 * 0.1f)) + 1.0f) * -4.0f;
+    }
 
-extern InitChainEntry D_80A676F4[];
+    if (this->unk_248 > 0) {
+        if (s->unk_00.unk_00 > 0) {
+            s->unk_00.unk_00--;
+        } else {
+            s->unk_00.unk_00 = Rand_S16Offset(30, 70);
+            s->unk_00.unk_04 = Rand_S16Offset(1000, 1000);
+            s->unk_00.unk_0C = (Rand_ZeroOne() * 1.5f) + 1.0f;
+            s->unk_00.unk_10 = fabsf(s->unk_00.unk_0C - s->unk_00.unk_08) * 0.05f;
+        }
+    } else {
+        s->unk_00.unk_00 = 0;
+        s->unk_00.unk_0C = phi_f2;
+        s->unk_00.unk_10 = (fabsf(s->unk_00.unk_0C - s->unk_00.unk_08) * 0.05f) + 0.02f;
+    }
 
-extern UNK_TYPE D_060016A0;
-extern UNK_TYPE D_06001AA8;
+    s->unk_00.unk_02 += s->unk_00.unk_04;
+    Math_StepToF(&s->unk_00.unk_08, s->unk_00.unk_0C, s->unk_00.unk_10);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Driftice/func_80A66570.s")
+    sp20 += (Math_SinS(s->unk_00.unk_02) * s->unk_00.unk_08);
+    return sp20;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Driftice/func_80A665AC.s")
+f32 func_80A667F0(ObjDrifticeStruct3* s, ObjDriftice* this) {
```

Comment:
We usually try to avoid names that short. I think it would be better to rename it to `arg0` instead